doc/vagrant.md

```diff
@@ -0,0 +1,75 @@
+# Using oi-userland with Vagrant
+
+oi-userland includes Vagrantfile, which allows contributing from non-OpenIndiana
+platforms such as Mac OS X, Linux or Windows. Currently, only the VirtualBox
+provider is supported!. Vagrant is suitable for cases when you don't have access to OpenIndiana host and want to develop/contribute to oi-userland.
+
+## Quickstart
+
+* Download Vagrant from the official [download site](https://www.vagrantup.com/downloads.html). Select a version for your platform. We recommend to use the latest version.
+* Download VirtualBox from the official [download site](https://www.virtualbox.org/wiki/Downloads). We recommend to  always use the latest VirtualBox.
+* Run the following commands:
+
+  ```
+  git clone https://github.com/OpenIndiana/oi-userland.git && \
+  cd oi-userland && \
+  vagrant up
+  ```
+
+## Getting started
+
+* Download Vagrant from the official [download site](https://www.vagrantup.com/downloads.html). Select a version for your platform. We recommend to use the latest version.
+* Download VirtualBox from the official [download site](https://www.virtualbox.org/wiki/Downloads). We recommend to use always the latest VirtualBox.
+* Fork the [oi-userland repository](https://github.com/OpenIndiana/oi-userland.git) on Github.
+* Clone your oi-userland repository somewhere:
+
+  ```
+  git clone https://github.com/<username>/oi-userland.git oi-userland
+  ```
+
+* Configure remote repository, so you can fetch updates easily:
+
+  ```
+  git remote add upstream https://github.com/OpenIndiana/oi-userland.git
+  ```
+
+  Updates are fetched with the following commands:
+
+  ```
+  git pull upstream oi/hipster
```

Comment:
use git pull --rebase. that removes the merge commits and makes it easier to make PR's against oi-userland